hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/FulltextSearchSvcImpl.java

```diff
@@ -382,4 +409,42 @@ public long count(String theResourceName, SearchParameterMap theParams) {
 
 		return queryOptionsStep.fetchTotalHitCount();
 	}
+
+	@Override
+	public List<IBaseResource> searchForResources(String theResourceType, SearchParameterMap theParams) {
+		int offset = 0; int limit = RSRC_SEARCH_LIMIT_DEFAULT;
+		if (theParams.getOffset() != null && theParams.getOffset() != 0) {
+			offset = theParams.getOffset();
+			limit = theParams.getCount();
+			// indicate param was already processed, otherwise queries DB to process it
+			theParams.setOffset(null);
+		}
+
+		mySearchCount++;
+
+		var query = getSearchSession().search(ResourceTable.class)
+				.select(this::buildResourceSelectClause)
+				.where(f -> buildWhereClause(f, theResourceType, theParams, null));
+
+		if (theParams.getSort() != null && offset == 0) {
+			query.sort(
```

Comment:
Shouldn't this be 
```suggestion
			query = query.sort(
```
Is this tested?

Reply:
No. Works as is. And it is tested now